src/main/java/org/reactome/release/cosmicupdate/COSMICUpdateUtil.java

```diff
@@ -0,0 +1,320 @@
+		logger.info("Now checking with CosmicFusionExport.tsv...");
+		try(CSVParser parser = new CSVParser(new FileReader(COSMICFusionExportFile), CSVFormat.DEFAULT.withFirstRecordAsHeader().withDelimiter('\t')); )
+		{
+			parser.forEach( record -> {
+				String fusionID = record.get(COSMIC_FUSION_ID);
+				if (fusionIDs.contains(fusionID))
+				{
+					updaters.get(COSMIC_FUSION_PREFIX+fusionID).setValid(true);
+				}
+			});
+		}
+		
+		logger.info("Now checking with CosmicMutationTracking.tsv...");
+		// Now we need to look through the HUGE file from COSMIC and see if we can map the identifiers...
+		try(CSVParser parser = new CSVParser(new FileReader(COSMICMutationTrackingFile), CSVFormat.DEFAULT.withFirstRecordAsHeader().withDelimiter('\t'));)
+		{
+			parser.forEach(record -> {
+				String legacyID = record.get(COSMIC_LEGACY_MUTATION_ID);
+				String mutationID = record.get(COSMIC_MUTATION_ID);
+				String genomicID = record.get(COSMIC_GENOMIC_MUTATION_ID);
+				if (updaters.containsKey(legacyID))
+				{
+					updaters.get(legacyID).getMutationIDs().add(mutationID);
+					updaters.get(legacyID).setCosvIdentifier(genomicID);
+				}
+			
+			});
+		}
+		logger.info("Now checking with CosmicMutantExport.tsv...");
+		try(CSVParser parser = new CSVParser(new FileReader(COSMICMutantExportFile), CSVFormat.DEFAULT.withFirstRecordAsHeader().withDelimiter('\t')); )
+		{
+			parser.forEach(record -> {
+				String legacyID = record.get(COSMIC_LEGACY_MUTATION_ID);
+				String mutationID = record.get(COSMIC_MUTATION_ID);
+				String genomicID = record.get(COSMIC_GENOMIC_MUTATION_ID);
+				if (updaters.containsKey(legacyID))
+				{
+					updaters.get(legacyID).setValid(true); // only VALID if in MutantExport...
+					updaters.get(legacyID).getMutationIDs().add(mutationID);
+					updaters.get(legacyID).setCosvIdentifier(genomicID);
+				}
+			});
+		}
+	}
+
+	/**
+	 * Determines the prefixes for COSMIC identifiers. The rule is:
+	 * IF an object has EWASes and there is an EWAS with a FragmentReplacedModification or a FragmentInsertionModification whose referenceSequence 
+	 * is NOT the referenceEntity of the EWAS... then the suggested prefix will be COSF (for Fusion), otherwise, COSM is suggested.
+	 * @param cosmicObjects Objects that are identified by a COSMIC identifier.
+	 * @return A map of <code>COSMICIdentifierUpdater</code>, keyed by COSMIC identifier.
+	 * @throws InvalidAttributeException
+	 * @throws Exception
+	 * @throws IOException
+	 */
+	static Map<String, COSMICIdentifierUpdater> determinePrefixes(Collection<GKInstance> cosmicObjects) throws InvalidAttributeException, Exception, IOException
+	{
+		Map<String, COSMICIdentifierUpdater> updates = new HashMap<>();
+		
+		// Create the reports directory if it's missing.
+		if (!Files.exists(Paths.get(COSMICUpdateUtil.reportsDirectoryPath)))
```

Comment:
This can be removed if you want,

```
public static Path createDirectories(Path dir,
                     FileAttribute<?>... attrs)
                              throws IOException

Creates a directory by creating all nonexistent parent directories first. Unlike the createDirectory method, an exception is not thrown if the directory could not be created because it already exists. 
```